homeassistant/components/whois/sensor.py

```diff
@@ -58,6 +65,24 @@ def _ensure_timezone(timestamp: datetime | None) -> datetime | None:
     return timestamp
 
 
+def _get_status_type(status: str | None) -> str | None:
+    """Get the status type from the status string.
+
+    Returns the status type in snake_case, so it can be used as a key for the translations.
+    E.g: "clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited" -> "client_delete_prohibited".
+    """
+    if status is None:
+        return None
+
+    # If the status is not in the STATUS_TYPES, return the status as is.
+    for icann_status, hass_status in STATUS_TYPES.items():
+        if icann_status in status:
+            return hass_status
+
```

Comment:
can you give an example of `status`? Because if it's just a string like `serverDeleteProhibited` or `clientTransferProhibited` without nothing more, this function can just be a `.get()`

Reply:
Yes, provided [here](https://github.com/home-assistant/core/pull/141051#discussion_r2011810190)